test/functional/p2p_post_invalidate_sync.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+# Copyright (c) 2014 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#
+# Test invalidateblock
+#
+
+from test_framework.test_framework import BitcoinTestFramework
+
+class InvalidateBlockTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 2
+        self.setup_clean_chain = True
+        self.extra_args = [['-debug', '-whitelist=127.0.0.1'],['-debug', '-whitelist=127.0.0.1']]
```

Comment:
why `-debug`? It's on by default.

Please comment why `-whitelist` is required

Reply:
Because that's what was there before, its just a test, and it doesn't seem to hurt anything.

Reply:
Here you go: https://www.python.org/dev/peps/pep-0257/

Reply:
How can it be there before if this is a new test?

Reply:
The code was taken, almost verbatim from #5806, its not fresh.

Reply:
Adding '-whitelist=127.0.0.1' to tests seems to me to make them less realistic, more brittle and harder to maintain if different connection options are added. It also changes p2p behaviour so that it's less obvious that the test is testing what you want to test in a focused way. Does whitelisting interact with how blocks are sync'ed in the event of a block being invalidated? I don't know, and by adding this option you're requiring reviewers to figure that out (or just rubberstamp ACK this PR without understanding the code). So my request is to either remove this or comment on why you've chosen to add this option.